### PR TITLE
Feat(eos_cli_config_gen): Add schema for ptp

### PR DESCRIPTION
**Pull request by mayurgs34**
### Contributor Checklist

- [x] Create schema fragment matching data model described in README.md and README_v4.0.md
  - README.md is most complete with all keys. README_v4.0 includes partial data models after conversion to lists.
  - Schema fragment path is `roles/eos_cli_config_gen/schemas/schema_fragments/<data_model_key>.schema.yml`.
  - Copy line 1-5 from another schema (comments and outer type:dict).
  - Refer to [schema documentation](https://avd.sh/en/devel/docs/input-variable-validation-BETA.html) for syntax and/or use YAML Lint plugin from Redhat in VSCode.
  - Use `convert_types` on value that could be mixed type or misinterpreted like integers and numeric strings.
- [x] If the data model has been converted from wildcard dicts:
  - Add `convert_types: ['dict']` to the schema.
  - Remove `convert_dicts` from the `templates/eos/<>.j2` and `templates/documentation/<>.j2` templates.
- [x] Run `molecule converge -s build_schemas_and_docs` to update schema and documentation.
- [x] Test by running `molecule converge -s eos_cli_config_gen` and verify no errors or changes to generated configs/docs.

### Reviewer Checklist

- Reviewer 1: Carl
  - [ ] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Claus
  - [x] Verify that data model is fully covered in the described schema. Easiest by looking at the generated documentation.
  - [x] Verify that `convert_dicts` has been removed from templates as applicable.
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+                    min: 2
+                    max: 255
+                    convert_types:
+                    - str
+              sequence_ids:
+                type: dict
+                keys:
+                  enabled:
+                    type: bool
+                    default: false
```
Comment (carlbuchmann) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]: Remove `default: false` as this is implied, and the ptp template will only configure if the value is defined.
```suggestion
```

**Review by carlbuchmann — CHANGES_REQUESTED** [review]
Small correction, otherwise LGTM!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml (hunk 1 above)
Comment (carlbuchmann) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]: resolved in d18407d

**Review by carlbuchmann — DISMISSED** [review]
LGTM!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  ptp:
+    type: dict
+    keys:
+      mode:
+        type: str
+        description: Mode value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        type: int
+        min: 0
+        max: 255
+        convert_types:
+        - str
+      ttl:
+        type: int
+        min: 1
+        max: 254
+        description: TTL value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        type: dict
+        keys:
+          general:
+            type: dict
+            keys:
+              dscp:
+                type: int
+                convert_types:
+                - str
+                description: DSCP value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+          event:
+            type: dict
+            keys:
+              dscp:
+                type: int
+                description: DSCP value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+            type: bool
+            default: true
+          threshold:
+            type: dict
+            keys:
+              offset_from_master:
+                type: int
+                min: 0
+                max: 1000000000
+                description: Offset value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+                convert_types:
+                - str
+              mean_path_delay:
+                type: int
+                min: 0
+                max: 1000000000
+                description: Delay value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+                convert_types:
+                - str
+              drop:
+                type: dict
+                keys:
+                  offset_from_master:
+                    type: int
+                    min: 0
+                    max: 1000000000
+                    description: Offset value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+                    convert_types:
+                    - str
+                  mean_path_delay:
+                    type: int
+                    min: 0
+                    max: 1000000000
+                    description: Delay value
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/ptp.schema.yml]:
```suggestion
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
These descriptions are the same as the key + "Value". I don't think we need this.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md [hunk 10]
```diff
@@ -2818,6 +2818,86 @@ prefix_lists:
 prompt: <str>
 ```
 
+## Ptp
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md]: we should also add PTP to the key_to_display_name.py

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/docs/Input Variables.md]: Carl will add it

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**